benchmark/utilities/openmm_utilities.py

```diff
@@ -86,43 +114,50 @@ def get_atoms_bonded_to_hydrogen(topology):
                 atom_indices.append(b.index)
             else:
                 atom_indices.append(a.index)
-    return atom_indices
+    return list(set(atom_indices))
 
 
 def repartition_hydrogen_mass_connected(topology, system, h_mass=4.0,
                                         mode="scale"  # or "decrement"
                                         ):
     """Set the mass of all hydrogens to h_mass. Reduce the mass of
     all atoms bonded to hydrogens, so that the total mass remains constant.
-
-    Question: how should we do this, exactly?
-    * Should we subtract the same mass from each bonded atom?
-    * Should we proportionally reduce the mass of each bonded atom?
     """
 
-    raise (NotImplementedError())
+    others = get_atoms_bonded_to_hydrogen(topology)
+    hydrogens = get_hydrogens(topology)
+    initial_h_mass = get_sum_of_masses(system, hydrogens) / len(hydrogens)
+    initial_mass_of_others = get_sum_of_masses(system, others)
+    mass_to_remove_from_others = (h_mass - initial_h_mass) * len(hydrogens)
+
+    if mode == "scale":
+        scale_factor = (initial_mass_of_others - mass_to_remove_from_others) / initial_mass_of_others
+        if scale_factor <= 0:
+            raise (RuntimeError("h_mass is too large! Can't remove this much mass from bonded atoms..."))
+
+        scale_particle_masses(system, others, scale_factor)
+
+    elif mode == "decrement":
```

Comment:
I thought the AMBER approach moved a specific predetermined quantity of mass to the H from the bonded heavy atom, but this implementation seems to move a quantity of mass that depends precisely on the composition of matter in the specific system.